### PR TITLE
Move near cache to datastore.

### PR DESCRIPTION
**Pull request by geekbeast**

### CODE REVIEW

**Review by katherinebernstein — APPROVED** [review]

File: conductor-client/src/main/kotlin/com/openlattice/hazelcast/pods/NearCachesPod.kt [hunk 1]
```diff
@@ -31,4 +31,17 @@ class NearCachesPod {
                 .setInMemoryFormat(InMemoryFormat.OBJECT)
                 .setMaxIdleSeconds(30)
     }
+
+    @Bean
+    fun permissionsNearcCacheConfig(): NearCacheConfig? {
```
Comment (UnsungHero97) [inline conductor-client/src/main/kotlin/com/openlattice/hazelcast/pods/NearCachesPod.kt:36]: Please remove extra `c` between `NearCache`.

**Review by UnsungHero97 — CHANGES_REQUESTED** [review]

**Review by makosblade — DISMISSED** [review]
So what exactly does this do aside from disabling NearCacheConfig on PermissionsMapstore? I don't see anything pulling in `NearCacheConfig` in this change or setting it on any maps

**Comment by geekbeast** [comment]
> So what exactly does this do aside from disabling NearCacheConfig on PermissionsMapstore? I don't see anything pulling in `NearCacheConfig` in this change or setting it on any maps

It moves the near cache to datastore instead of conductor.

Configuration happens inside of our standard hazelcast instance configuration routines.

https://github.com/geekbeast/rhizome/blob/bb8a0740f24f8f1e147dd4c29af3fb15cd6cd357/src/main/java/com/kryptnostic/rhizome/pods/hazelcast/RegistryBasedHazelcastInstanceConfigurationPod.java#L115
https://github.com/geekbeast/rhizome/blob/bb8a0740f24f8f1e147dd4c29af3fb15cd6cd357/src/main/java/com/kryptnostic/rhizome/pods/hazelcast/BaseHazelcastInstanceConfigurationPod.java#L135

**Comment by makosblade** [comment]
> > So what exactly does this do aside from disabling NearCacheConfig on PermissionsMapstore? I don't see anything pulling in `NearCacheConfig` in this change or setting it on any maps
> 
> It moves the near cache to datastore instead of conductor.
> 
> Configuration happens inside of our standard hazelcast instance configuration routines.
> 
> https://github.com/geekbeast/rhizome/blob/bb8a0740f24f8f1e147dd4c29af3fb15cd6cd357/src/main/java/com/kryptnostic/rhizome/pods/hazelcast/RegistryBasedHazelcastInstanceConfigurationPod.java#L115
> https://github.com/geekbeast/rhizome/blob/bb8a0740f24f8f1e147dd4c29af3fb15cd6cd357/src/main/java/com/kryptnostic/rhizome/pods/hazelcast/BaseHazelcastInstanceConfigurationPod.java#L135

Okay cool. It seems like this sets up a NearCache config for *all* mapstores, not just the PermissionsMapstore though. Is that intended?

**Comment by geekbeast** [comment]
> > > So what exactly does this do aside from disabling NearCacheConfig on PermissionsMapstore? I don't see anything pulling in `NearCacheConfig` in this change or setting it on any maps
> > 
> > 
> > It moves the near cache to datastore instead of conductor.
> > Configuration happens inside of our standard hazelcast instance configuration routines.
> > https://github.com/geekbeast/rhizome/blob/bb8a0740f24f8f1e147dd4c29af3fb15cd6cd357/src/main/java/com/kryptnostic/rhizome/pods/hazelcast/RegistryBasedHazelcastInstanceConfigurationPod.java#L115
> > https://github.com/geekbeast/rhizome/blob/bb8a0740f24f8f1e147dd4c29af3fb15cd6cd357/src/main/java/com/kryptnostic/rhizome/pods/hazelcast/BaseHazelcastInstanceConfigurationPod.java#L135
> 
> Okay cool. It seems like this sets up a NearCache config for _all_ mapstores, not just the PermissionsMapstore though. Is that intended?

Woops nice catch. I updated to only be on permission map.

**Review by katherinebernstein — APPROVED** [review]

**Review by UnsungHero97 — APPROVED** [review]

**Review by makosblade — APPROVED** [review]

**Pull request merged**